docusaurus/docs/ui-extensions/register-an-extension.md

```diff
@@ -118,6 +118,88 @@ new AppPlugin().configureExtensionLink({
 });
 ```
 
+### Example: Display a React component
+
+Using React components as extensions is a powerful way to extend the functionality of an existing UI either in core Grafana or in an other plugin.
+
+```ts title="src/module.ts"
+new AppPlugin().configureExtensionComponent({
+  title: 'My component extension',
+  description: 'Renders a button at a pre-defined extension point.',
+  extensionPointId: 'grafana/<extension-point-id>',
+  component: MyExtension,
+});
+```
+
+```tsx title="src/components/MyExtension.tsx"
+export const MyExtension = () => (
+  <Button onClick={() => {}} variant="secondary" type="button">
+    Extend UI
+  </Button>
+);
+```
+
+### Example: Use the context of the extension point in a component
+
+Extension points can pass contextual information to the extensions they render, we refer to this data as `context`.
+In case it is a core extension point (defined in core Grafana), there is also always a type definition available for these context objects. In the following example we will cover how to use this `context` in our component extension and also get the static types right.
```

Comment:
```suggestion
Extension points defined in core Grafana additionally have a type definition available for these context objects. In the following example we will cover how to use this `context` in our component extension and also get the static types right.
```

Reply:
not sure on this, but a suggestion

Reply:
I think it's a good suggestion, will commit it now. (Joe will confirm it later I guess.)